test/integration/modules/edgecore/eventbus/eventbus_test.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"database/sql"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/kubeedge/kubeedge/test/integration/utils/common"
+
+	MQTT "github.com/eclipse/paho.mqtt.golang"
+	_ "github.com/mattn/go-sqlite3"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+//DeviceUpdate device update
+type DeviceUpdate struct {
+	State string `json:"state,omitempty"`
+}
+
+//the struct of device
+type Device struct {
+	id          []uint8
+	name        string
+	description string
+	state       string
+	lastOnline  []uint8
+}
+
+//Function to access the edge_core DB and return the device state.
+func getDeviceStateFromDB(deviceId string) string {
+	var device Device
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		common.Failf("Failed to get PWD: %v", err)
+		os.Exit(1)
+	}
+	destpath := filepath.Join(pwd, "../../edge.db")
+	db, err := sql.Open("sqlite3", destpath)
+	if err != nil {
+		common.Failf("Open Sqlite DB failed :%v", err)
+	}
+	defer db.Close()
+	rows, err := db.Query("SELECT * FROM device", 0)
```

Comment:
we can optimize the select call by adding a where clause since we passing in the deviceID.
```go
rows, err := db.Query("SELECT * FROM device where deviceID= ?", deviceID)
```